Upgrade EUI to v48.1.1

breehall opened the pull request:
## Summary

eui@46.1.0 ⏩ eui@48.1.1

## [`48.1.1`](https://github.com/elastic/eui/tree/v48.1.1)

- Added a `data-test-subj` attribute to `EuiDataGrid` cell expansion buttons ([#5643](https://github.com/elastic/eui/pull/5643))
- Added a `data-test-selected` attribute to `EuiSelectable` list options ([#5643](https://github.com/elastic/eui/pull/5643))

## [`48.1.0`](https://github.com/elastic/eui/tree/v48.1.0)

- Improved `EuiSelectable` keypress scenarios ([#5613](https://github.com/elastic/eui/pull/5613))
- Converted `FieldValueSelectionFilter` in `EuiSearchBar` to use `EuiSelectable` ([#5387](https://github.com/elastic/eui/issues/5387))
- Added `lineDashed`, `lineDotted`, and `lineSolid` glyphs to `EuiIcon` ([#5633](https://github.com/elastic/eui/pull/5633))
- Updated `colorMode` return type in theme context utilities ([#5639](https://github.com/elastic/eui/issues/5639))

## [`48.0.0`](https://github.com/elastic/eui/tree/v48.0.0)

- Refactored `EuiSuggest` to use `EuiSelectable` ([#5157](https://github.com/elastic/eui/pull/5157))
- Added a return type to `EuiTable` `resolveWidthAsStyle` util ([#5615](https://github.com/elastic/eui/pull/5615))
- Added a return type to `euiSelectableTemplateSitewideFormatOptions` util ([#5620](https://github.com/elastic/eui/pull/5620))

**Bug fixes**

- Fixed `EuiDataGrid` to correctly remove the cell expansion action button when a column sets both `cellActions` and `isExpandable` to false ([#5592](https://github.com/elastic/eui/pull/5592))
- Fixed `EuiDataGrid` re-playing the cell actions animation when hovering over an already-focused cell ([#5592](https://github.com/elastic/eui/pull/5592))
- Fixed `EuiDataGrid` auto row heights bugging out when cell popovers are opened ([#5622](https://github.com/elastic/eui/pull/5622))

**Breaking changes**

- Removed `EuiSuggestInput` ([#5157](https://github.com/elastic/eui/pull/5157))
- Required `aria-label` or `aria-labelledby` for `EuiSuggest` ([#5157](https://github.com/elastic/eui/pull/5157))
- Renamed `sendValue` prop to `onSearchChange` for `EuiSuggest` ([#5157](https://github.com/elastic/eui/pull/5157))

## [`47.0.0`](https://github.com/elastic/eui/tree/v47.0.0)

- Added support for React 17 (this is a backwards compatible change - React 16.12+ is still supported for consuming applications) ([#5584](https://github.com/elastic/eui/pull/5584))
- Added the ability to control internal `EuiDataGrid` fullscreen, cell focus, and cell popover state via the `ref` prop ([#5590](https://github.com/elastic/eui/pull/5590))
- Added `paddingSize` prop to `EuiSelectableList` ([#5581](https://github.com/elastic/eui/pull/5581))
- Added `errorMessage` prop to `EuiSelectable` ([#5581](https://github.com/elastic/eui/pull/5581))
- Refactored `EuiSelectable` accessibility ([#5581](https://github.com/elastic/eui/pull/5581))
- Updated `tokenTag` design to look more like a tag ([#5553](https://github.com/elastic/eui/pull/5553))
- Lowered border radius of `.euiToken--square` to look more like a square ([#5553](https://github.com/elastic/eui/pull/5553))

**Bug fixes**

- Fixed `EuiInMemoryTable`'s `onTableChange` callback not returning the correct `sort.field` value on pagination ([#5588](https://github.com/elastic/eui/pull/5588))
- Fixed `EuiFilePicker` allowing files to be removed when `disabled` ([#5603](https://github.com/elastic/eui/pull/5603))

**Breaking changes**

- Upgraded TypeScript version to ~4.5.3 ([#5591](https://github.com/elastic/eui/pull/5591))

## [`46.2.0`](https://github.com/elastic/eui/tree/v46.2.0)

- Updated `EuiDataGrid`s with scrolling content to always have a border around the grid body and any scrollbars ([#5563](https://github.com/elastic/eui/pull/5563))
- Updated `EuiDataGrid`'s body to a light gray background, which primarily shows when scrolling through virtualized content ([#5562](https://github.com/elastic/eui/pull/5562))
- Added referenceable `id` for the generated label in `EuiFormRow` ([#5574](https://github.com/elastic/eui/pull/5574))
- Addeed optional attribute configurations in `EuiPopover` to aid screen reader announcements ([#5574](https://github.com/elastic/eui/pull/5574))
- Added `ref` passthroughs to `EuiIputPopover` subcomponents ([#5574](https://github.com/elastic/eui/pull/5574))
- Added `EuiScreenReaderLive` component for updateable `aria-live` regions ([#5567](https://github.com/elastic/eui/pull/5567))

**Bug fixes**

- Fixed an accessibility issue where `EuiComboBoxPill` close button had a nested interactive element ([#5560](https://github.com/elastic/eui/pull/5560))
- Fixed EuiDataGrid height issue when in full-screen mode and with scrolling content ([#5557](https://github.com/elastic/eui/pull/5557))
- Fixed an accessibility issue in custom and interactive Drag and Drop patterns ([#5568](https://github.com/elastic/eui/pull/5568))
- Fixed a focus bug in `EuiDataGrid` when clicking another cell header with an already-open cell header popover ([#5556](https://github.com/elastic/eui/pull/5556))
- Fixed `EuiDataGrid` to always focus back into the grid on pagination ([#5587](https://github.com/elastic/eui/pull/5587))
- Fixed `EuiDataGrid` and `EuiTable` pagination potentially rendering out view on narrow tables with many pages ([#5561](https://github.com/elastic/eui/pull/5561))

breehall committed:
Upgraded the version of EUI to 47.0.0 and react-beautiful-dnd (an EUI dependancy) to 13.1.0

breehall committed:
Update i18n mappings with required changes

cee-chen committed:
Fix all data grid types/tests missing the new `colIndex` prop passed by renderCellValue and cellActions

cee-chen committed:
Fix cellActions closePopover type to indicate conditionality (only passed when popover is open)

cee-chen committed:
Fix more datagrid colIndex errors

- pass more missing `colIndex`s

- pass RowAction colIndex, because it's inheriting types from EUI

- omit colIndex from the leading controls column renderer, because it doesn't need them

cee-chen committed:
Improve StatefulCell typing

- pass colIndex (which fixes EUI type match issue)
- DRY out ariaColIndex logic
- rename ariaRowindex passed to StatefulCell to rowIndex

breehall committed:
Updated i18n_eui_mapping tests to add euiSelectable.searchResults to the tokensToSkip array

breehall committed:
Merge branch 'eui-upgrade-47.0.0' of https://github.com/elastic/kibana into eui-upgrade-47.0.0
Merging in latest commits from eui-upgrade-47.0.0

cee-chen committed:
Fix failing ML datagrid FTR test

- `visibleRowIndex` is not a prop passed back by `popoverContents` (see `EuiDataGridCellValueElementProps`), but `rowIndex` is

breehall committed:
Merge branch 'eui-upgrade-47.0.0' of https://github.com/elastic/kibana into eui-upgrade-47.0.0
Merging in the latest code from eui-upgrade-47.0.0

constancecchen commented:
Removing the v7.17.1 label, we haven't backported any recent upgrades to 7.x recently

constancecchen commented:
Oh, 8.1 FF was a week or two ago, so we should probably change that as well. We don't need this to be on 8.1, correct @chandlerprall @thompsongl?

chandlerprall commented:
8.2 is the right target 👍

cee-chen committed:
Revert attempted ML type change

- in actuality this is an EUI typing issue, children is a ReactElement, not a ReactNode. However we'll shortly be deprecationg popoverContents, so this isn't worth fixing right now

breehall committed:
Merge branch 'eui-upgrade-47.0.0' of https://github.com/elastic/kibana into eui-upgrade-47.0.0
Pulling in the latest code from eui-upgrade-47.0.0

breehall committed:
Update test files to include up to date snapshots of code samples

breehall committed:
Updated a test snapshots to match the latest version of code samples

breehall committed:
Upgraded the version of EUI from 47.0.0 to 48.0.0 in package.json and license_checker config files

breehall committed:
Update the required i18n translation mapping file with additions and changes from EUI version 48.0.0

breehall committed:
Updated three security screen accessibility tests to check for the aria-checked attribute instead of the aria-selected attribute as part of an accessibility update to aria made in EUI PR 5581

breehall committed:
Updated two unit cases to that are responsible for checking strict equality of strings. These unit tests were for the EuiSelectable and EuiFilterGroup components. Both of these components contain and utilize EuiScreenReaderOnly which provides text that is used for screen readers, but can still be viewed and queried in the DOM. These tests have been updated with the EuiScreenReaderOnly text in mind.

breehall committed:
Code clean up and added a missing internationalization token

breehall committed:
Merged in the latest code from the main branch and resolved a merge conflic inside of yarn.lock

constancecchen commented:
@breehall Rather than fixing the yarn.lock merge conflict, what you'll likely need to do is rerun `yarn kbn bootstrap` locally and push up the new yarn.lock file it outputs

breehall commented:
> @breehall Rather than fixing the yarn.lock merge conflict, what you'll likely need to do is rerun `yarn kbn bootstrap` locally and push up the new yarn.lock file it outputs

I just realized that. `yarn kbn boostrap` is running locally now.

breehall committed:
Ran yarn kbn bootstrap to update the yarn lock file

cee-chen committed:
Fix failing ML FTR test
- EuiSelectable now relies on aria-checked to indicate selected state, per W3 spec

cee-chen committed:
Fix failing functional tests that click the datagrid cell expand button

.euiDataGridRowCell__expandButtonIcon was deprecated in favor of a shared .euiDataGridRowCell__actionButtonIcon class, but the expand action is always the last one

breehall committed:
Merge branch 'main' into eui-upgrade-47.0.0

elasticmachine commented:
Pinging @elastic/eui-design (EUI)

spalger commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-upgrade-47.0.0

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

kibanamachine committed:
Merge branch 'main' into eui-upgrade-47.0.0

cee-chen committed:
Upgrade to 48.1.1

cee-chen committed:
Switch to data-test-subj for datagrid cell expansion selectors

cee-chen committed:
Switch to new `data-test-selected` attribute over `aria-checked`

lukeelmers reviewed (approved):
Core mapping changes LGTM

mistic reviewed (approved):
Changes on files under operations team code owners LGTM

cee-chen committed:
Update snapshots/Jest tests to account for EuiSelectable use in EuiFilter

kertal reviewed (approved):
DataDiscovery.team owned code LGTM, code review, only tests were changed. didn't test

kertal commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-upgrade-47.0.0

kertal commented:
think it's convenient to deploy EUI upgrades to cloud, for people that wanna also test it but don't need to deploy locally, added the tag `c:deploy-cloud`, and here we go: https://kibana-pr-125023.kb.us-west2.gcp.elastic-cloud.com 
this is so fantastic, kudos again to @elastic/kibana-operations

jloleysens reviewed (approved):
AppServices LGTM, snap files changed. Did not test new version locally.

mbondyra reviewed (approved):
VisEditors changes LGTM 👌🏼

kibanamachine committed:
Merge branch 'main' into eui-upgrade-47.0.0

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-upgrade-47.0.0

MadameSheema commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-upgrade-47.0.0

alisonelizabeth reviewed (approved):
platform deployment management changes LGTM

breehall committed:
Updated the eui-upgrade-47.0.0 branch with code from main and ran kbn bootsrap to resolve a merge conflict in the yarn.lock file

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-upgrade-47.0.0

breehall commented:
Merging this PR with the current reviews as it has been open for review for about a week and is passing all CI tests.

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 125023` or prevent reminders by adding the `backport:skip` label.
